app/models/tipline_message.rb

```diff
@@ -6,6 +6,8 @@ class TiplineMessage < ApplicationRecord
   validates_presence_of :team, :uid, :platform, :language, :direction, :sent_at, :payload, :state
   validates_inclusion_of :state, in: ['sent', 'received', 'delivered']
 
+  after_commit :verify_user_rate_limit, on: :create
```

Comment:
I'm not familiar with ruby, would we want to verify the rate limit before the commit?

Reply:
Ah, it's because once a user is blocked we don't even process their incoming request, so we won't even attempt to save their message.